### ISSUE TITLE
Add typescript support

### ISSUE THREAD

**Issue by dabbeg** [HermannBjorgvin/Kennitala#10]
Hi it would be awesome if you could typescript support for kennitala

**Comment by HermannBjorgvin** [comment]
I have 0 experience with TypeScript so I don't feel comfortable as of now to be publishing TypeScript packages. I'm also concerned about the distribution size but that might not be a big factor. I'd be interested in some way to support TypeScript but I don't have plans right now.

I'm open to collaboration and pull requests, it's just a question of how to implement it, distribution size, watch out for breaking changes.

Tryggingarmiðstöðinn has a TypeScript Kennitölu library that's pretty popular with a similar feature set but it doesn't have exactly the same API as I do. https://github.com/tryggingamidstodin/kennitala-utility

**Comment by dabbeg** [comment]
Okay I will try to find time to make a PR, the distribution size should almost be the same. But I will note the new size in the PR.

Yeah I saw the library from Tryggingarmiðstöðinn but I like the API of this library better.

**Comment by HermannBjorgvin** [comment]
That's great! Good luck with it

The only real requirements are maintaining the same support for older node.js versions, up to a reasonable point. We can check out the unit tests in Travis CI and see which node.js versions still work in your PR no problem. Should be easy but just a thing to watch out for with packages, better to transpile to older versions than to break support

**Comment by HermannBjorgvin** [comment]
Friend of mine sent me this declaration file for the library. I'm not totally sure how this works with TypeScript when you import a package but the minified code does not have the same variable names but it has a source map. It's something I haven't researched yet.

Do you want the package with a declaration file or are you thinking about a refactor? A declaration might be enough since the library is small and very well vetted by now.

```javascript
declare module "kennitala" {
    export function isValid(kennitala: string): boolean;
    export function isPerson(kennitala: string): boolean;
    export function isCompany(kennitala: string): boolean;

    export function clean(kennitala: string): string;
    export function format(kennitala: string, spacer: string?): string;

    export function generatePerson(date: Date): string;
    export function generateCompany(date: Date): string;

    export function info(kennitala: string): string;

    export interface KennitalaInfo {
        kt: string;
        valid: boolean;
        type: "person" | "company";
        birthday: Date;
        birthdayReadable: string;
        age: number;
    }
}
```

**Comment by dabbeg** [comment]
Awesome! Yes this solves my problem :smiley: 
It would be nicer to refactor but as there is not much activity going on here I think this is good enough.

**Comment by HermannBjorgvin** [comment]
Issue closed in #12 

NPM package has been published as v1.2.4

No breaking changes or code changes but added type declaration file to enable TypeScript support.
Also improved API docs slightly.
